crates/stc_ts_file_analyzer/src/analyzer/function/mod.rs

```diff
@@ -478,6 +480,25 @@ impl Analyzer<'_, '_> {
     }
 }
 
+// Check for reserved words used as identifiers (yield, await, etc.)
+//
+// TODO(@littledivy): Generalize this to work regardless of 'function' context.
+fn validate_parameter_name(ctx: &mut Analyzer, p: &RParam, is_async: bool, is_generator: bool) {
```

Comment:
This should be made context-aware so it can be used with in top level and not just function params. Maybe in the parser? I saw @kdy1 that you have a similar `TODO` a few lines above.

Similar to https://github.com/microsoft/TypeScript/blob/3d2b4017eb6b9a2b94bc673291e56ae95e8beddd/src/compiler/binder.ts#L2178

Reply:
I think it's better to make a method of `Analyzer` for consistency.
I may refactor it to split the analyzer, but let's use a method at now